### DISCUSSION TITLE
mpremote disabling RTC?

### DISCUSSION THREAD

**Discussion by robhamerling** [micropython/micropython#13034] [m.iMX RT / Teensy 4.x]
I experience a non-running RTC after copying files to the Teensy40.
After loading a new firmware version I restore files to the Teensy40 with a script with a series of 'mpremote cp :' commands.  So far that worked fine, but I recently noticed that after this script RTC is not running anymore. 
Immediately after loading new firmware (today 1.22.0-preview.164) I checked (and confirmed) that RTC is running, but after my copy script not anymore. But it seems programs run fine, e.g. a simple blink-a-led program using time.sleep() show a blinking LED at expected frequency. However time.gmtime() keeps returning the same values.  So it looks like mpremote does something 'special' to RTC.  What could be the cause of the standstill of RTC?

**Comment by jimmo** [comment]
hi @robhamerling I wasn't able to repro this on a Teensy 4.0

For example:

```
$ mpremote rtc
(2013, 10, 14, 0, 19, 53, 18, 0)
$ mpremote rtc --set
$ mpremote rtc
(2023, 11, 21, 1, 11, 53, 24, 0)
$ mpremote rtc
(2023, 11, 21, 1, 11, 53, 26, 0)
$ mpremote fs cp ~/mpy/foo.py :
cp /home/jimmo/mpy/foo.py :
$ mpremote rtc
(2023, 11, 21, 1, 11, 53, 31, 0)
$ mpremote rtc
(2023, 11, 21, 1, 11, 53, 33, 0)
```

Can you give a sequence of steps I can run to demonstrate what you're seeing?

I looked at the code too, the obvious thing would be that `mpremote` triggers a soft reset, but all that does is remove the RTC alarm, it doesn't stop the RTC.

**Comment by robert-hh** [comment]
@robhamerling We had a discussion about the RTC of your Teensy in September about RTC stopping with a power cycle. A firmware change caused the RTC to survive a power cycle. Still, i could not reproduce the behavior with my devices here.

**Comment by robhamerling** [comment]
Ah, I found the cause of the RTC standstill! **Not mpremote!**
In my main.py I call a script to sync RTC from a DS3231 module. When the DS3231 module could not be detected that script initializes RTC with an invalid value.  This causes RTC to stop running.
For an example see the terminal log below. 
Although this is a bug in my script, it is strange that it stops the time.
```
>>> from machine import RTC
>>> rtc=RTC()
>>> rtc.datetime()
(2023, 11, 21, 1, 12, 42, 35, 0)
>>> rtc.datetime()
(2023, 11, 21, 1, 12, 42, 36, 0)
>>> rtc.datetime()
(2023, 11, 21, 1, 12, 42, 38, 0)

>>> rtc.datetime((-1,-1,-1,-1,-1,-1,-1,-1))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError:
>>> rtc.datetime()
(2000, 11, 21, 1, 0, 0, 28, 0)
>>> rtc.datetime()
(2000, 11, 21, 1, 0, 0, 28, 0)
>>> rtc.datetime()
(2000, 11, 21, 1, 0, 0, 28, 0)
>>>

```

**Reply by robert-hh** [reply]
> it is strange that it stops the time

Luckily it only stops the clock.
